### PR TITLE
Unify test modes in databricks test script [databricks]

### PR DESCRIPTION
**Pull request by pxLi**
Signed-off-by: Peixin Li <pxli@nyu.edu>

a follow-up of #6008, to unify the test modes in test scripts

### CODE REVIEW

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
iceberg test failed in databricks 10.4 runtime https://github.com/NVIDIA/spark-rapids/issues/6022

File: jenkins/databricks/test.sh [hunk 1]
```diff
@@ -69,10 +69,9 @@ IS_SPARK_311_OR_LATER=0
 
 
 # TEST_MODE
-# - IT_ONLY
-# - CUDF_UDF_ONLY
-# - ALL: IT+CUDF_UDF
-TEST_MODE=${TEST_MODE:-'IT_ONLY'}
+# - DEFAULT: all tests except cudf_udf tests
+# - CUDF_UDF_ONLY: cudf_udf tests only, requires extra conda cudf-py lib
```
Comment (tgravescs) [inline jenkins/databricks/test.sh:73]: seems like we should add the iceberg and delta lake as comments here

Comment (pxLi) [reply jenkins/databricks/test.sh:73]: yeah, I will update this one

**Review by jlowe — APPROVED** [review]

**Review by tgravescs — APPROVED** [review]

**Pull request merged**